buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -400,13 +392,9 @@ private static void shutdownExecutorService() {
         }
     }
 
-    private static void shutDownAllClusters() {
-        if (logger.isDebugEnabled()) {
-            logger.debug("Shutting down all test clusters", new RuntimeException());
-        }
+    private void shutDownAllClusters() {
```

Comment:
If we keep the JVM shutdown hook I don't think there's anything to prevent this from being called twice and in-turn, potentially calling `stop()` on our clusters multiple times.

Reply:
that should be fine because we still synchronize on `runningClusters` so there might be multiple calls to `shutDownAllClusters` but not to `stop` on the actual clusters

Reply:
Synchonizing `runningClusters` won't help if we never remove them from the collection. We call `stop()` but we never actually clear out this collection so the subsequent call will call `stop()` again.